clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -234,10 +234,17 @@ public RuntimeException lastError() {
     }
 
     public synchronized boolean ensurePartitionAdded(TopicPartition tp) {
```

Comment:
Would it be too much trouble to add a couple basic unit tests just for this function?